File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    idx1=0
    idx2=0
    for i in goal:
        if idx1<len(cards1) and cards1[idx1]==i:
            idx1+=1
            continue
        elif idx2<len(cards2) and cards2[idx2]==i:
            idx2+=1
            continue
        else:
            return 'No'
    return 'Yes'

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        boolean check = true;
        int idx1 = 0;
        int idx2 = 0;
        for(String s : goal){
            if(idx1 < cards1.length && s.equals(cards1[idx1])){
                idx1++;
            }
            else if(idx2 < cards2.length && s.equals(cards2[idx2])){
                idx2++;
            }
            else{
                check = false;
                break;
            }
        }
        return check ? "Yes" : "No";
    }
}